python/xorbits/_mars/dataframe/reduction/aggregation.py

```diff
@@ -1021,13 +1059,25 @@ def aggregate(df, func=None, axis=0, **kw):
     output_type = kw.pop("_output_type", None)
     dtypes = kw.pop("_dtypes", None)
     index = kw.pop("_index", None)
+    raw_func_name = kw.pop("_func_name", None)
 
     if not is_funcs_aggregate(func, func_kw=kw, ndim=df.ndim):
         return df.transform(func, axis=axis, _call_agg=True)
 
+    func_rename = (
```

Comment:
Can you explain the purpose of this variable?

Reply:
The principle of dataframe reduction is to convert all aggregation functions into specific func objects and pass them into agg. However, cudf requires that a string must be passed in to the agg method, so it is necessary to obtain the string representation of the aggregation function. The "func_rename" here represents the string representation of the specific aggregation method.

For example, for .agg('sum'), the func obj passed to the agg function is DataFrameSum.get_reduction_callable.sum_.
If I just use func.__name__, I will get "sum_" rather than "sum".

So I get the real func name string from DataFrameSum._func_name. Therefore, I need to mark the func name by func_rename in the case of GPU.

Reply:
> Can you explain the purpose of this variable?

The principle of dataframe reduction is to convert all aggregation functions into specific func objects and pass them into agg. However, cudf requires that a string must be passed in to the agg method, so it is necessary to obtain the string representation of the aggregation function. The "func_rename" here represents the string representation of the specific aggregation method.

For example, for .agg('sum'), the func obj passed to the agg function is DataFrameSum.get_reduction_callable.sum_.
If I just use func.name, I will get "sum_" rather than "sum".

So I get the real func name string from DataFrameSum._func_name. Therefore, I need to mark the func name by func_rename in the case of GPU.

Reply:
Correct me if I'm wrong. When `aggregate` is called by a user, `raw_func_name` will be `None` and func could be `callable`, `str`, or `list[callable | str]`.

Do you need to handle this scenario?

Reply:
Understood. I think,

1. The name of `func_rename` is a little bit confusing since we are not actually rename it. As I understand it, this variable stores the function names (it's a list), would it be more readable if we just name it `func_names`?
2. If a `callable` is passed as the func, cudf will definitly raise an exception, would it be better to raise the exception here instead of letting it crash during execution?

Reply:
> 1. The name of `func_rename` is a little bit confusing since we are not actually rename it. As I understand it, this variable stores the function names (it's a list), would it be more readable if we just name it `func_names`?
> 2. If a `callable` is passed as the func, cudf will definitly raise an exception, would it be better to raise the exception here instead of letting it crash during execution?

1. function_rename is originally a field on the op, I think there is no need to add a new one. Secondly, in fact, the semantics of rename here is because the function passed in by the user (whether it is func or str) is first converted into a func object through ``reduction`` logic, and I am expressing it as a string here. 

2. Even if the user passes a string, cudf does not support all agg functions, like pandas, which means that we need to extract all unsupported ones and provide an error report? In addition, there is a change in the error message in this PR that can indicate that there is a NotImplError. In addition, another advantage is that when cudf supports certain methods in the future, we will naturally support them.

What do you think @aresnow1 ?

Reply:
`func_rename` seems an attribute that record raw names, it's confusing but acceptable for now, the reduction implementation is quite complex and need refactor in the future.
About question2, we can catch errors in `execute` and tell user to use string as aggregation function.